paddle/optimizer/serialization.h

```diff
@@ -0,0 +1,36 @@
+#ifndef PADDLE_OPTIMIZER_SERIALIZARION_H
```

Comment:
agreed. But since this is almost a finished PR, add simple serialization feature for saving checkpoint. Which will be developed into a new PR alone.